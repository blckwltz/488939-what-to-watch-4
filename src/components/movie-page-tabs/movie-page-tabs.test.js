import React from 'react';
import renderer from 'react-test-renderer';
import MoviePageTabs from './movie-page-tabs';

const mocks = {
  title: `Bohemian Rhapsody`,
  genre: `Biography, Music`,
  releaseDate: 2018,
  runTime: `2h 50m`,
  cover: `img/the-grand-budapest-hotel-poster.jpg`,
  poster: `img/bohemian-rhapsody.jpg`,
  previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  rating: {
    score: 8.0,
    level: `Good`,
    count: 240,
  },
  description: `The story of the legendary British rock band Queen and lead singer Freddie Mercury,
      leading up to their famous performance at Live Aid (1985).`,
  director: `Bryan Singer`,
  cast: `Rami Malek, Lucy Boynton, Gwilym Lee`,
  reviews: [
    {
      author: `Kate Muir`,
      text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious
          Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
      date: `2016-12-24`,
      rating: 8.7,
    },
    {
      author: `Bill Goodykoontz`,
      text: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes
          I wish I could take back.`,
      date: `2015-11-18`,
      rating: 5.3,
    },
  ],
};

it(`Should render Tabs component correctly`, () => {
  const tree = renderer
    .create(
        <MoviePageTabs movieInfo={mocks}/>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
