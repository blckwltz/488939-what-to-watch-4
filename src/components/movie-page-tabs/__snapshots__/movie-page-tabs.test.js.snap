// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Tabs component correctly 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item false"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item false"
        onClick={[Function]}
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Good
      </span>
      <span
        className="movie-rating__count"
      >
        240
         ratings
      </span>
    </p>
  </div>
  <div
    className="movie-card__text"
  >
    <p>
      The story of the legendary British rock band Queen and lead singer Freddie Mercury,
      leading up to their famous performance at Live Aid (1985).
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Bryan Singer
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Rami Malek, Lucy Boynton, Gwilym Lee
         and other
      </strong>
    </p>
  </div>
</div>
`;
